proxy/pkg/command/server.go

```diff
@@ -250,10 +251,24 @@ func Server(cfg *config.Config) *cli.Command {
 }
 
 func loadMiddlewares(ctx context.Context, l log.Logger, cfg *config.Config) alice.Chain {
-	accountsClient := acc.NewAccountsService("com.owncloud.api.accounts", grpc.DefaultClient)
+
 	rolesClient := settings.NewRoleService("com.owncloud.api.settings", grpc.DefaultClient)
-	storeClient := storepb.NewStoreService("com.owncloud.api.store", grpc.DefaultClient)
 	revaClient, err := cs3.GetGatewayServiceClient(cfg.Reva.Address)
+	var userProvider provider.UserProvider
+	switch cfg.AccountBackendType {
+	case "accounts":
+		userProvider = provider.NewAccountsServiceProvider(
+			acc.NewAccountsService("com.owncloud.api.accounts", grpc.DefaultClient),
+			rolesClient,
+			cfg.OIDC.Issuer,
+		)
+	case "cs3":
+		userProvider = provider.NewCS3UserProvider(revaClient, rolesClient, revaClient)
+	default:
+		l.Fatal().Msgf("Invalid accounts backend type '%s'", cfg.AccountBackendType)
```

Comment:
wouldn't we want to default to the accounts service if none is present?

Reply:
We do this implicity by flagset. If an incorrect value is provided it should fail loudly.